### ISSUE TITLE
Bond angle deviations after refinement

### ISSUE THREAD

**Issue by helgepat** [keitaroyam/servalcat#8]
We have observed that the refined models through servalcat contain more bond angle deviations than when Phenix is used. We check our models with Phenix validation, so I am not sure if this is a biased analysis towards Phenix?
This is an example from Phenix validation:
![grafik](https://user-images.githubusercontent.com/97705752/206135361-e363c432-c2f9-4939-9cd8-65506850afc6.png)

Many of these angles are 114° when they should be 120°?

Is there something we are doing wrong or could be doing better with our modelling?

**Comment by keitaroyam** [comment]
Thanks for reporting! In the CCP4 monomer library (CCP4-ML), phosphodiester bond related angles are defined like this (in $CLIBD_MON/list/mon_lib_list.cif):
```
loop_
_chem_link_angle.link_id
_chem_link_angle.atom_1_comp_id
_chem_link_angle.atom_id_1
_chem_link_angle.atom_2_comp_id
_chem_link_angle.atom_id_2
_chem_link_angle.atom_3_comp_id
_chem_link_angle.atom_id_3
_chem_link_angle.value_angle
_chem_link_angle.value_angle_esd
p 1 "C3'" 1 "O3'" 2 P 109.471 3.00
p 2 OP1 2 P 1 "O3'" 108.530 2.45
p 2 OP2 2 P 1 "O3'" 108.530 2.45
p 2 "O5'" 2 P 1 "O3'" 101.077 2.30
```
So C3'-O3'-P angle is 109.5±3°, while phenix seems to use 120.2±1.5°. In this case, their value looks more accurate. Perhaps these ideal angles were set long ago and has not been updated (109.471° looks like too ideal sp3 angle and not from actual chemical knowledge). We will update the CCP4-ML, but in the meantime, you can use this acedrg-updated file [p_link_revised.zip](https://github.com/keitaroyam/servalcat/files/10175483/p_link_revised.zip) with `--ligand` option.

I cannot see atom names in bond lengths outliers. Could you expand atom name columns if worried?

**Comment by helgepat** [comment]
That is good to hear, I was wondering if I did something wrong in Coot ;)

Here is a picture for the expanded bond length values:
![grafik](https://user-images.githubusercontent.com/97705752/206179680-26af7740-4639-438d-94e1-89b9e593aa26.png)

**Comment by keitaroyam** [comment]
Thanks. I checked some of them in CCP4-ML.
C: C1'-N1 1.487±0.01 Å, N1-C2 1.397±0.01 Å
G: C1'-N9 1.458±0.01 Å, O5'-C5' 1.450±0.0166 Å, 

They seem to be more or less consistent with phenix values (G's O5'-C5' is actually a bit different though). I would first check these atoms with high deviations visually. There may be something causing these strains (e.g. too close contact, out of density). Or you may just need to tighten the restraint weight.

**Comment by helgepat** [comment]
We tested now with the restraint you provided and indeed it fixes the issue after updating the model in Coot with the new restraints. Thank you for the help, yet again! 
I noticed that there might be another discrepancy between the Phenix and Refmac restraints (less abundant than the first one), there are a lot of 97° in the model when it should be 104° ?
![grafik](https://user-images.githubusercontent.com/97705752/207069747-36119720-974c-4407-8db7-81aea7bc1e59.png)

**Comment by keitaroyam** [comment]
I see the major problem is O3'-P-O5' angle, of which ideal value in phenix is 104.0±1.5° and in the dictionary I gave 100.7±3.0°. So deviation for 97.961° is ~1σ in "ours" but 4σ in phenix. I suppose this is not so problematic, but I will look into it.

If you want, you can change sigma value in the dictionary. Find in p_link_revised.cif
```
loop_
_chem_link_angle.link_id
_chem_link_angle.atom_1_comp_id
_chem_link_angle.atom_id_1
_chem_link_angle.atom_2_comp_id
_chem_link_angle.atom_id_2
_chem_link_angle.atom_3_comp_id
_chem_link_angle.atom_id_3
_chem_link_angle.value_angle
_chem_link_angle.value_angle_esd
p 1 "C3'" 1 "O3'" 2 P 121.082 1.50 
p 2 OP1 2 P 1 "O3'" 109.493 3.00 
p 2 OP2 2 P 1 "O3'" 109.493 3.00 
p 2 "O5'" 2 P 1 "O3'" 100.661 3.00 
```
and change 3.00 at the bottom to 1.5 for example.

**Comment by helgepat** [comment]
Thanks again for the reply, after your explanation I will leave these as they are. I don't really know how to check these values myself, thats why its sometimes surprising that there are differences between software packages. I like the table from Phenix validate to directly navigate to problematic regions thats why I keep coming back to it. 
With servalcat I like the visual aspect of identifying problems through the difference map, but its also easy to get lost in a corner of the ribosome and miss some issues further away. Therefore I like the table format from Phenix validate... of course its always a question of what to fix from the provided list especially if Refmac handles stuff differently...

**Comment by keitaroyam** [comment]
Yes, comparing with other package is a good practice, and we actually found the C3'-O3'-P angle problem. I agree we need a similar mechanism (GUI). Validate - Geometry analysis in Coot is perhaps more Refmac compatible way (at least it uses the same monomer library, if you are using Coot installed with CCP4). By the way, Refmac by default only reports outliers with >10 sigma (which are copied in servalcat log also).

You may see even more outliers in PDB validation, as they often use smaller sigma values. I would not consider such outliers problematic unless ideal value in the dictionary is wrong. But it is always good to check outliers with density, and we may see how to fix it.

**Comment by helgepat** [comment]
I am in the process of re-refining my models with the new angle restraints. Are these now included in the newest CCP4 release (CCP4 8.0.007)?

Meanwhile, I was wondering: Is there are standalone command to merge .cif restraints without running servalcat?

**Comment by keitaroyam** [comment]
Unfortunately not (I could do it if I knew when update happened..)

You can merge dictionary cif files:
```sh
servalcat util merge_dicts 1.cif 2.cif ... [-o merged.cif]
```
